### PR TITLE
Python tests for MinGW

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -32,15 +35,25 @@ install:
 - if [%COMPILER%]==[MinGW] rename "C:\Program Files (x86)\Git\bin\sh.exe" "sh2.exe"
 - if [%COMPILER%]==[MinGW] mingw-get install mingw32-gmp
 
+- if [%WITH_PYTHON%]==[yes] set PATH=C:\Python%PYTHON_VERSION%\Scripts;%PATH%
+- if [%WITH_PYTHON%]==[yes] pip install cython nose pytest sympy
```
Comment (certik) [inline appveyor.yml]: The pytest doesn't seem to be needed, since we only test nosetests?

File: cmake/FindCython.cmake [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 # This finds the "cython" executable in your PATH, and then in some standard
 # paths:
-FIND_FILE(CYTHON_BIN cython /usr/bin /usr/local/bin)
+FIND_PROGRAM(CYTHON_BIN cython /usr/bin /usr/local/bin)
```
Comment (certik) [inline cmake/FindCython.cmake]: I think this is the right change, but why was it failing before?

Comment (isuruf) [inline cmake/FindCython.cmake]: `FIND_FILE` finds a file while `FIND_PROGRAM` finds an executable. When `cython.exe` is in the path it is found by `FIND_PROGRAM`

**Comment by certik** [comment]
This looks good. Both Travis and AppVeyor tests pass, merging.

File: cmake/FindCython.cmake (hunk 2 above)
Comment (certik) [inline cmake/FindCython.cmake]: I see, but `cython.exe` is not found by `find_file`, is that right?

Comment (isuruf) [inline cmake/FindCython.cmake]: Yes

**Comment by certik** [comment]
Only the flint lib tests times out, but that is fixed by #551.

**Comment by certik** [comment]
I am going to merge #551 first, since it affects all our tests. Then I merge this one.

File: cmake/FindCython.cmake (hunk 2 above)
Comment (isuruf) [inline cmake/FindCython.cmake]: I was wrong. Both are essentially the same, but the name we are searching for is `cython`. `find_file` sees `cython.exe` and ignores it while `find_program` finds it.

**Comment by certik** [comment]
I restarted that one failing test. It should pass now.

**Comment by certik** [comment]
The Python 3 tests fail because our cmake finds `python` which is Python 2, so it links the wrong library. The fix should be to add the right version of Python into Path, or perhaps add some options to our cmake to pick up the right Python.

**Comment by isuruf** [comment]
Modified the path to pick up the correct python version

**Comment by certik** [comment]
This looks good.

The only comment I have is that I think we perhaps only need to test Python 3.4? Because we also need to test Python with MSVC, and also 32 vs 64 bit. Maybe we can do 2.7 and 3.4. I wouldn't do more.

**Comment by certik** [comment]
+1 to merge

**Pull request merged**